runners/spark/src/main/java/org/apache/beam/runners/spark/translation/DoFnFunction.java

```diff
@@ -93,6 +93,13 @@ public synchronized void output(WindowedValue<OutputT> o) {
     }
 
     @Override
+    public <T> void sideOutput(TupleTag<T> tag, WindowedValue<T> output) {
+      throw new UnsupportedOperationException(
+          "sideOutput is an unsupported operation for doFunctions, use a "
+              + "MultiDoFunction instead.");
+    }
+
```

Comment:
This is a complementing comment to say that `sideOutputWithTimestamp` should throw `UnsupportedOperationException` as well.